homeassistant/components/tado/climate.py

```diff
@@ -598,21 +609,23 @@ def _async_update_zone_data(self) -> None:
                     self._device_id
                 ][TEMP_OFFSET][offset_key]
 
-        self._current_tado_fan_speed = (
-            self._tado_zone_data.current_fan_level
-            if self._tado_zone_data.current_fan_level is not None
-            else self._tado_zone_data.current_fan_speed
-        )
-
         self._current_tado_hvac_mode = self._tado_zone_data.current_hvac_mode
         self._current_tado_hvac_action = self._tado_zone_data.current_hvac_action
-        self._current_tado_swing_mode = self._tado_zone_data.current_swing_mode
-        self._current_tado_vertical_swing = (
-            self._tado_zone_data.current_vertical_swing_mode
-        )
-        self._current_tado_horizontal_swing = (
-            self._tado_zone_data.current_horizontal_swing_mode
-        )
+
+        if self._does_current_hvac_mode_have_fan_level():
```

Comment:
Where are they made as dedicated functions? Maybe a more generic function can be used? This is seems to clutter a bit. :)

Reply:
All right, I remove this function to be more generic.